src/api/TaskCollaborators.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @flow
+ * @file Helper for the box Task Assignments API
+ * @author Box
+ */
+
+import Base from './Base';
+import {
+    ERROR_CODE_FETCH_TASK_COLLABORATOR,
+    ERROR_CODE_CREATE_TASK_COLLABORATOR,
+    ERROR_CODE_UPDATE_TASK_COLLABORATOR,
+    ERROR_CODE_DELETE_TASK_COLLABORATOR,
+    API_PAGE_LIMIT,
+} from '../constants';
+
+class TaskCollaborators extends Base {
+    getUrlForTaskCollaborators(taskId: string): string {
+        return `${this.getBaseApiUrl()}/undoc/tasks/${taskId}/task_collaborators?limit=${API_PAGE_LIMIT}`;
+    }
+
+    getUrlForTaskCollaboratorCreate(): string {
+        return `${this.getBaseApiUrl()}/undoc/task_collaborators`;
+    }
+
+    getUrlForTaskCollaborator(id: string): string {
+        return `${this.getBaseApiUrl()}/undoc/task_collaborators/${id}`;
+    }
+
+    createTaskCollaborator({
+        errorCallback,
+        file,
+        successCallback,
+        task,
+        user,
+    }: {
+        errorCallback: ElementsErrorCallback,
+        file: BoxItem,
+        successCallback: Function,
+        task: { id: string },
+        user: { id: string },
+    }): void {
+        this.errorCode = ERROR_CODE_CREATE_TASK_COLLABORATOR;
+
+        const requestData = {
```

Comment:
Should you check permissions here, from the file object, so that edits aren't allowed if permission is not set?
Likewise for other sections?

Reply:
That logic has been moved into the UI code
- we want to prevent this check from being called by disabling UI elements instead of bouncing the request
- we also expect to handle auth errors resulting from server side auth checks for anything that can't be computed beforehand